tests/src/pod-smoke.spec.ts

```diff
@@ -0,0 +1,177 @@
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { Page } from 'playwright';
+import type { RunnerTestContext } from './testContext/runner-test-context';
+import { afterAll, beforeAll, test, describe, beforeEach, expect } from 'vitest';
+import { expect as playExpect } from '@playwright/test';
+import { PodmanDesktopRunner } from './runner/podman-desktop-runner';
+import { WelcomePage } from './model/pages/welcome-page';
+import { NavigationBar } from './model/workbench/navigation';
+import { waitUntil, waitWhile } from './utility/wait';
+import { deleteContainer, deleteImage, deletePod } from './utility/operations';
+import { ContainerState, PodState } from './model/core/states';
+
+let pdRunner: PodmanDesktopRunner;
+let page: Page;
+
+const imageToPull = 'ghcr.io/linuxcontainers/alpine';
+const imageTag = 'latest';
+const containerToRun = 'alpine-container';
+const podToRun = 'alpine-pod';
+
+beforeAll(async () => {
+  pdRunner = new PodmanDesktopRunner();
+  page = await pdRunner.start();
+  pdRunner.setVideoName('pods-e2e');
+  const welcomePage = new WelcomePage(page);
+  await welcomePage.handleWelcomePage(true);
+  // wait giving a time to podman desktop to load up
+  const images = await new NavigationBar(page).openImages();
+  await waitWhile(
+    async () => await images.pageIsEmpty(),
+    5000,
+    1000,
+    false,
+    'Images page is empty, there are no images present',
+  );
+  await deletePod(page, podToRun);
+  await deleteContainer(page, containerToRun);
+});
+
+beforeEach<RunnerTestContext>(async ctx => {
+  ctx.pdRunner = pdRunner;
+});
+
+afterAll(async () => {
+  await deletePod(page, podToRun);
+  await deleteContainer(page, containerToRun);
+  await deleteImage(page, imageToPull);
+  await pdRunner.close();
+}, 90000);
+
+describe.skipIf(process.env.GITHUB_ACTIONS && process.env.RUNNER_OS === 'Linux')(
+  'Verification of pod creation workflow',
+  async () => {
+    test(`Pulling of '${imageToPull}:${imageTag}' image`, async () => {
+      const navigationBar = new NavigationBar(page);
+      let images = await navigationBar.openImages();
+      const pullImagePage = await images.openPullImage();
+      images = await pullImagePage.pullImage(imageToPull, imageTag, 180000);
```

Comment:
the timeout for pulling an image is a way too huge. also, if we happen to encounter such timeoutwhen pulling an image, it will be limited to a test timeout, in this case 60 sec. So we wont even hae a chance for such timeout.

Reply:
Ok, reducing time to pull to 60s.